Prototope/Layer.swift

```diff
@@ -253,13 +258,23 @@ public class Layer: Equatable, Hashable {
 	}
 
 	/** The scaling factor of the layer along the x dimension. Defaults to 1. */
-	public var scaleX: Double = 1 {
-		didSet { updateTransform() }
+	public var scaleX: Double {
+        get {
+            return layer.valueForKeyPath("transform.scale.x") as Double
```

Comment:
I totally see what you mean, unfortunately I'm not sure we can hide UIKit completely. A parallel example to yours:
Say you animate a layer's rotation, and sometime during the animation you query the value, hoping to get the *current* rotation. You'd get the final value.

I'm not sure how we can effectively provide both values, while keeping the API simple and not exposing such CA details :/ But I haven't thought about this enough.